scripts/feature_tests/upgrade/combined_tests/1cp_1w_bootDiskImageANDK8sControllers_clusterLevel_upgrade.sh

```diff
@@ -13,84 +13,101 @@ echo '' >~/.ssh/known_hosts
 
 start_logging "${1}"
 # Provision original nodes
-set_number_of_master_node_replicas 1
+set_number_of_master_node_replicas 3
 set_number_of_worker_node_replicas 1
 
 provision_controlplane_node
 
-controlplane_is_provisioned
-controlplane_has_correct_replicas 1
+set_kubeconfig_towards_target_cluster
+controlplane_has_correct_replicas 3
 
-# apply CNI
-apply_cni
+point_to_management_cluster
 
 provision_worker_node
+point_to_target_cluster
 worker_has_correct_replicas 1
 
-# ----------- upgrade controlplane components ---------------
-cleanup_clusterctl_configuration
+point_to_management_cluster
 
-buildClusterctl
+#export IMAGE_RAW_NAME="UBUNTU_NODE_IMAGE_K8S_v1.18.8-raw.img"
 
-# Install initial version folder structure
-pushd /tmp/cluster-api-clone || exit
-cmd/clusterctl/hack/create-local-repository.py
+# Start pivoting
+pushd "${METAL3_DEV_ENV_DIR}/scripts/feature_tests/pivoting" || exit
+make upgrading
 popd || exit
+sleep 120
+#
+point_to_target_cluster
+#
+scale_controlplane_to 1
+controlplane_has_correct_replicas 1
 
-create_clusterctl_configuration
-
-createNextVersionControllers
-
-makeCrdChanges
-
-# show upgrade plan
-clusterctl upgrade plan
-# do upgrade
-clusterctl upgrade plan | grep "upgrade apply" | xargs | xargs clusterctl
-# shellcheck disable=SC2082
-# Verify upgrade
-upgraded_controllers_count=$(kubectl api-resources | grep -Ec "kcp2020|ma2020")
-upgraded_bootstrap_crd_count=$(kubectl get crds \
-  kubeadmconfigs.bootstrap.cluster.x-k8s.io -o json | jq '.spec.names.singular' | wc -l)
-upgraded_capm3_controller_count=$(kubectl api-resources | grep -c m3c2020)
-
-if [ "${upgraded_controllers_count}" -ne 2 ]; then
-  log_error "Failed to upgrade cluster-api and controlplane components"
-  log_test_result "1cp_1w_bootDiskImageANDK8sCotrollers_clusterLevel_upgrade.sh" "fail"
-  exit 1
-fi
-if [ "${upgraded_bootstrap_crd_count}" -ne 1 ]; then
-  log_error "Failed to upgrade control-plane-kubeadm components"
-  log_test_result "1cp_1w_bootDiskImageANDK8sCotrollers_clusterLevel_upgrade.sh" "fail"
-  exit 1
-fi
-
-if [ "${upgraded_capm3_controller_count}" -ne 1 ]; then
-  log_error "Failed to upgrade infrastructure components"
-  log_test_result "1cp_1w_bootDiskImageANDK8sCotrollers_clusterLevel_upgrade.sh" "fail"
-  exit 1
-fi
-
-sleep 30 # Wait for the controllers to be up and running
-
-for i in {1..10}; do
-  health_controllers=$(kubectl get pods -A | grep -E "capm3-system|capi-kubeadm|metal3" |
-    grep -vc 'Running')
-  if [ "${health_controllers}" -eq 0 ]; then
-    break
-  fi
-  if [[ "${i}" -ge 10 ]]; then
-    log_error "Some of the upgraded controlplane components are not healthy"
-    log_test_result "1cp_1w_bootDiskImageANDK8sCotrollers_clusterLevel_upgrade.sh" "fail"
-    exit 1
-  fi
-  sleep 10
-done
+# Untaint the masters
+manage_node_taints
```

Comment:
manage_node_taints  function need to be applied in line 42. Otherwise all the pods from controlplane will move to the worker.